pkg/reconciler/openshift/tektontrigger/extension.go

```diff
@@ -18,28 +18,72 @@ package tektontrigger
 
 import (
 	"context"
+	"os"
+	"path/filepath"
 
+	"github.com/go-logr/zapr"
+	mfc "github.com/manifestival/client-go-client"
 	mf "github.com/manifestival/manifestival"
 	"github.com/tektoncd/operator/pkg/apis/operator/v1alpha1"
+	"github.com/tektoncd/operator/pkg/client/clientset/versioned"
+	operatorclient "github.com/tektoncd/operator/pkg/client/injection/client"
 	"github.com/tektoncd/operator/pkg/reconciler/common"
+	occommon "github.com/tektoncd/operator/pkg/reconciler/openshift/common"
+	"go.uber.org/zap"
+	"knative.dev/pkg/injection"
+	"knative.dev/pkg/logging"
 )
 
 // NoPlatform "generates" a NilExtension
-func OpenShiftExtension(context.Context) common.Extension {
-	return openshiftExtension{}
+func OpenShiftExtension(ctx context.Context) common.Extension {
+	logger := logging.FromContext(ctx)
+	mfclient, err := mfc.NewClient(injection.GetConfig(ctx))
+	if err != nil {
+		logger.Fatalw("error creating client from injected config", zap.Error(err))
+	}
+	mflogger := zapr.NewLogger(logger.Named("manifestival").Desugar())
+	manifest, err := mf.ManifestFrom(mf.Slice{}, mf.UseClient(mfclient), mf.UseLogger(mflogger))
+	if err != nil {
+		logger.Fatalw("error creating initial manifest", zap.Error(err))
+	}
+	ext := openshiftExtension{
+		operatorClientSet: operatorclient.Get(ctx),
+		manifest:          manifest,
+	}
+	return ext
 }
 
-type openshiftExtension struct{}
+type openshiftExtension struct {
+	operatorClientSet versioned.Interface
+	manifest          mf.Manifest
+}
 
 func (oe openshiftExtension) Transformers(comp v1alpha1.TektonComponent) []mf.Transformer {
 	triggerImages := common.ToLowerCaseKeys(common.ImagesFromEnv(common.TriggersImagePrefix))
 	return []mf.Transformer{
 		common.DeploymentImages(triggerImages),
+		occommon.ApplyCABundles,
 	}
 }
-func (oe openshiftExtension) PreReconcile(context.Context, v1alpha1.TektonComponent) error {
-	return nil
+
+func (oe openshiftExtension) PreReconcile(ctx context.Context, tc v1alpha1.TektonComponent) error {
+	koDataDir := os.Getenv(common.KoEnvKey)
+
+	// make sure that openshift-pipelines namespace exists
+	namespaceLocation := filepath.Join(koDataDir, "operator-namespace.yaml")
+	if err := common.AppendManifest(&oe.manifest, namespaceLocation); err != nil {
+		return err
+	}
+
+	// add inject CA bundles manifests
+	cabundlesLocation := filepath.Join(koDataDir, "cabundles")
+	if err := common.AppendManifest(&oe.manifest, cabundlesLocation); err != nil {
+		return err
+	}
+
+	return common.Install(ctx, &oe.manifest, tc)
```

Comment:
we can remove this duplication, as Triggers won't install if pipeline is not installed
https://github.com/tektoncd/operator/blob/6b4b0a872d36eaadb0e02d0a542fdfc303cab586/pkg/reconciler/kubernetes/tektontrigger/tektontrigger.go#L109

Reply:
we don't necessarily need this extension in triggers. Infact, we can move this extension to TektonConfig Reconciler.
@savitaashture @vdemeester

Reply:
PTAL https://github.com/tektoncd/operator/pull/246#discussion_r592202582

Reply:
i see..

```
common.DeploymentImages(triggerImages),
		occommon.ApplyCABundles,
``` 
is required, but the implementation of prereconcile could be just `{return nil}`

Reply:
fixed